deployment/upgradeLive

```diff
@@ -2,43 +2,133 @@
+    cp /opt/kaltura/liveController/configMapping.ini /opt/kaltura/liveController/$VERSION/liveRecorder/Config/
     npm install
     cd ..
     # download binaries from Jenkins
     echo "### starting to download binaries"
-    bash /opt/kaltura/liveController/$1/deployment/get_bins "$1"
+    /opt/kaltura/liveController/$VERSION/deployment/get_bins "$VERSION"
     echo "### finished downloading binaries"
     # create symlinks to latest dir and process script
-	if [ -L "latest" ]; then
-		echo "unlink latest"
-		unlink latest
-	fi
-    ln -s /opt/kaltura/liveController/$1 /opt/kaltura/liveController/latest
-	if [ ! -L /etc/init.d/kLiveController ]; then
-		echo "Creating symlink to kLiveController script at - /etc/init.d/kLiveController"
-		ln -s /opt/kaltura/liveController/latest/serviceWrappers/linux/kLiveController /etc/init.d/kLiveController
-	fi
+    if [ -L "latest" ]; then
+         echo "unlink latest"
+         unlink latest
+    fi
+    ln -s /opt/kaltura/liveController/$VERSION /opt/kaltura/liveController/latest
+
+}
+
+function downgrade() {
+    if [ -L "latest" ]; then
+        echo "unlink latest"
+        unlink latest
+    fi
+    ln -s /opt/kaltura/liveController/$VERSION /opt/kaltura/liveController/latest
+
+}
+
+function stop() {
+   /etc/init.d/kLiveController stop
+   if [ -r /etc/init.d/LiveRecorder ]; then
+        /etc/init.d/LiveRecorder stop
+   fi
+}
+
+function start() {
+    if [ ! -L /etc/init.d/kLiveController ]; then
+            echo "Creating symlink to kLiveController script at - /etc/init.d/kLiveController"
+            ln -s /opt/kaltura/liveController/latest/serviceWrappers/linux/kLiveController /etc/init.d/kLiveController
+    fi
     # start nginx (web server), liveController and liveRecorder
-    /etc/init.d/kLiveController restart
+    echo "starting liveController..."
+    /etc/init.d/kLiveController start
     sleep 5
-    /opt/kaltura/liveController/$1/packager/bin/run_nginx.sh
-    /etc/init.d/liveRecorder restart
+    echo "starting nginx..."
+    /opt/kaltura/liveController/$VERSION/packager/bin/run_nginx.sh
+    if [ -r /etc/init.d/LiveRecorder ]; then
+          if [ ! -L /etc/init.d/LiveRecorder ]; then
+                echo "Creating symlink to LiveRecorder script at - /etc/init.d/LiveRecorder"
+                ln -s /opt/kaltura/liveController/latest/liveRecorder/serviceWrappers/linux/LiveRecorder /etc/init.d/LiveRecorder
+          fi
+          echo "starting liveRecorder..."
+          /etc/init.d/liveRecorder start
+    else
+        echo "this machine doesn't run liveRecorder"
+    fi
+}
+
+if [ ! -d "$VERSION" ] || [ "$PARAM" != "install" ] ; then
+
+    case $PARAM in
+        install)
+            echo "upgrading liveController to v${VERSION}"
+            stop
+            download_new_version
+            install
+            start
+            ;;
+        replace_script)
+            echo "replacing upgradeLive script to v${VERSION}"
+            replace_upgrade_script $VERSION
```

Comment:
what's the meaning of the variable at the end of the line?
$VERSION is defined above and is know by all functions